Crunch keyring tests.R

```diff
@@ -0,0 +1,113 @@
+keyring::key_delete("crunch")
```

Comment:
I think we should break out the logic around getting a username password combination into a separate function so that it is easier to test in the existing automated testing infrastructure.

I'm thinking lines 45-74 from auth.R get wrapped in a function like:

```r
get_user_pass_combo <- function(email, pass) {
    # fill in with lines from 45-74
    # ...

    return(list(email = email, pass = pass))
}
```

Then we call that function in `login()`.

This will make testing easier, because we can just check that the `get_user_pass_combo()` gets the username and email we expect in different situations.


Then we'll need to fit your tests into the existing infrastructure. We use `testthat`, and a good explainer is in [Hadley Wickham's R packages book](http://r-pkgs.had.co.nz/tests.html), though we have some of our own specialness baked in.

I think these should go in the `tests/testthat/test-auth.R` file, around here: https://github.com/Crunch-io/rcrunch/blob/automation-script/tests/testthat/test-auth.R#L4
(And that test should be updated now that your changes make passing NULL allowed)

You can run just this file by running `devtools::test(filter = "auth")` from R or `make test file='auth'` from the command line.